src/Build.UnitTests/Evaluation/Expander_Tests.cs

```diff
@@ -3609,6 +3609,38 @@ public void PropertyStringConstructorConsumingItemMetadata(string metadatumName,
             result.ShouldBe(metadatumValue);
         }
 
+        [Fact]
+        public void PropertyFunctionHashCodeSameOnlyIfStringSame()
+        {
+            PropertyDictionary<ProjectPropertyInstance> pg = new PropertyDictionary<ProjectPropertyInstance>();
+            Expander<ProjectPropertyInstance, ProjectItemInstance> expander = new Expander<ProjectPropertyInstance, ProjectItemInstance>(pg, FileSystems.Default);
+            int[] hashes = {
+                (int)expander.ExpandPropertiesLeaveTypedAndEscaped("$([MSBuild]::StableStringHash('cat1s'))", ExpanderOptions.ExpandProperties, MockElementLocation.Instance),
+                (int)expander.ExpandPropertiesLeaveTypedAndEscaped("$([MSBuild]::StableStringHash('cat1z'))", ExpanderOptions.ExpandProperties, MockElementLocation.Instance),
+                (int)expander.ExpandPropertiesLeaveTypedAndEscaped("$([MSBuild]::StableStringHash('bat1s'))", ExpanderOptions.ExpandProperties, MockElementLocation.Instance),
+                (int)expander.ExpandPropertiesLeaveTypedAndEscaped("$([MSBuild]::StableStringHash('cut1s'))", ExpanderOptions.ExpandProperties, MockElementLocation.Instance),
+                (int)expander.ExpandPropertiesLeaveTypedAndEscaped("$([MSBuild]::StableStringHash('cat1so'))", ExpanderOptions.ExpandProperties, MockElementLocation.Instance),
+                (int)expander.ExpandPropertiesLeaveTypedAndEscaped("$([MSBuild]::StableStringHash('cats1'))", ExpanderOptions.ExpandProperties, MockElementLocation.Instance),
+                (int)expander.ExpandPropertiesLeaveTypedAndEscaped("$([MSBuild]::StableStringHash('acat1s'))", ExpanderOptions.ExpandProperties, MockElementLocation.Instance),
+                (int)expander.ExpandPropertiesLeaveTypedAndEscaped("$([MSBuild]::StableStringHash('cat12s'))", ExpanderOptions.ExpandProperties, MockElementLocation.Instance),
+                (int)expander.ExpandPropertiesLeaveTypedAndEscaped("$([MSBuild]::StableStringHash('cat1s'))", ExpanderOptions.ExpandProperties, MockElementLocation.Instance)
```

Comment:
nit: Perhaps you could create an array with just the strings to hash (`"cat1s", "cat1z", ...`) to eliminate code/text duplication.